crucible-llvm/src/Lang/Crucible/LLVM/MemModel/Common.hs

```diff
@@ -244,11 +244,18 @@ rangeLoad lo ltp s@(R so se)
  where le = typeEnd lo ltp
        loadFail = ValueCtorVar (OutOfRange lo ltp)
 
+-- | Produces a @Mux ValueCtor@ expression representing the range load conditions
+-- when the load and store offsets are concrete and the store size is bounded
 fixedOffsetRangeLoad :: Addr
+                     -- ^ Address of load
                      -> StorageType
+                     -- ^ Type to load
                      -> Addr
+                     -- ^ Address of store
                      -> Mux (ValueCtor (RangeLoad Addr Addr))
 fixedOffsetRangeLoad l tp s
+  | le < l = -- the load crosses a boundary
```

Comment:
OK, we need to think about this situation a bit more.  Right now, we're using `Word64` to represent concrete addresses, and relying on wrap-around behavior of those bakes in a 64-bit assumption on the memory model.

So, we should figure out how to generalize this the right way, and also see if we need to additionally update the symbolic load operations.

Reply:
Changing Bytes to natural numbers means that this extra logic is not necessary any more